src/safeposix/syscalls/net_calls.rs

```diff
@@ -1576,10 +1580,19 @@ impl Cage {
                     return syscall_error(Errno::EINVAL, "epoll wait", "max events argument is not a positive number");
                 }
                 let mut poll_fds_vec: Vec<PollStruct> = vec![];
+                let mut rm_fds_vec: Vec<i32> = vec![];
                 let mut num_events: usize = 0;
                 for set in epollfdobj.registered_fds.iter() {
                     let (&key, &value) = set.pair();
 
+                    // check if any of the registered fds were closed, add them to remove list
+                    let checkedregfd = self.get_filedescriptor(key).unwrap();
+                    let unlocked_regfd = checkedregfd.read();
```

Comment:
I would suggest Error Handling with unwrap: Using unwrap() can cause panics. It's preferable to handle potential errors gracefully.